src/logging/WriterFrontend.h

```diff
@@ -8,6 +8,70 @@ namespace zeek::logging {
 
 class Manager;
 
+
+namespace detail {
+
+/**
+ * Implements a buffer accumulating log records in \a WriterFrontend instance
+ * before passing them to \a WriterBackend instances.
+ *
+ * \see WriterFrontend::Write
+ */
+class WriteBuffer {
+public:
+    /**
+     * Constructor.
+     */
+    explicit WriteBuffer(size_t buffer_size) : buffer_size(buffer_size) {}
+    /**
+     * Push a record to the buffer.
+     *
+     * @param record The records vals.
+     */
+    void PushRecord(LogRecord&& record) { records.emplace_back(std::move(record)); }
```

Comment:
Suggestion on terminology: Given that this a *write* buffer, maybe call this either `WriteRecord` or even simply `Write`?